### PR TITLE
Fix statically linked concurrency on Linux

### PR DESCRIPTION
**Pull request by etcwilde**
Generating a statically-linked executable either with
`-static-executable` or `-static-stdlib` that contains concurrency needs
to link the concurrency libraries or the missing symbols will cause link
failures.

This patch adds dispatch and blocks runtime to the list of statically
linked libraries. In the case of the static stdlib, it only adds them if
the concurrency mechanisms use dispatch, otherwise it doesn't.

For the static executable, it always adds them since that doesn't appear
to be very configurable.

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

**Review by ktoso — APPROVED** [review]

File: stdlib/public/runtime/CMakeLists.txt [hunk 1]
```diff
@@ -209,17 +209,21 @@ foreach(sdk ${SWIFT_SDKS})
...
+        set(concurrency_libs "-ldispatch -lBlocksRuntime")
       endif()
 
       set(linkfile ${lowercase_sdk}/static-stdlib-args.lnk)
       file(WRITE "${SWIFTSTATICLIB_DIR}/${linkfile}" "
 -ldl
 ${libpthread}
 ${android_libraries}
+${concurrency_libs}
 -lswiftCore
```
Comment (compnerd) [inline stdlib/public/runtime/CMakeLists.txt:227]: Hmm, please swap L227 and L226.  IIRC, with Concurrency, swiftCore can make a reference to libdispatch.

**Review by compnerd — APPROVED** [review]
LGTM with the small change.

File: stdlib/public/runtime/CMakeLists.txt (hunk 1 above)
Comment (etcwilde) [reply stdlib/public/runtime/CMakeLists.txt:226]: Done.

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
Broken by https://github.com/apple/swift/pull/42073

**Comment by etcwilde** [comment]
@swift-ci please test macOS

**Comment by etcwilde** [comment]
@swift-ci please test Windows

**Comment by etcwilde** [comment]
https://github.com/apple/swift/pull/42075 has now merged, which should hopefully fix things again.

**Comment by etcwilde** [comment]
@swift-ci please test macOS

**Pull request merged**